api/app/controllers/api/v1/ingestions_controller.rb

```diff
@@ -44,6 +54,32 @@ def update
         )
       end
 
+      def reset
+        @ingestion = load_ingestion
+        @ingestion.reset
+        @ingestion.save
+      end
+
+      def process_ingestion
+        @ingestion = load_ingestion
+        Ingestions::ProcessIngestionJob.perform_later(@ingestion.id, current_user)
+      end
+
+      def reingest
+        @ingestion = load_ingestion
+        Ingestions::ReingestIngestionJob.perform_later(@ingestion.id, current_user)
+      end
+
+      def show_messages
+        @ingestion = load_ingestion
+        permitted = params.permit(:starting_at)
+        time = DateTime.parse(permitted[:starting_at])
+        render_multiple_resources IngestionMessage.where(
```

Comment:
In addition to moving this to an `ingestions/relationships/ingestion_messages_controller.rb`, I think it would be best to just have FE retrieve all the messages at once (via `skip_pagination`) and let them decide if they need to show anything extra, rather than paginating by time.